spring-cloud-loadbalancer/src/main/java/org/springframework/cloud/client/loadbalancer/impl/scoped/CachedDiscoveryClient.java

```diff
@@ -0,0 +1,31 @@
+package org.springframework.cloud.client.loadbalancer.impl.scoped;
+
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.cloud.client.ServiceInstance;
+import org.springframework.cloud.client.discovery.DiscoveryClient;
+import org.springframework.core.env.Environment;
+
+import java.util.List;
+
+/**
+ * @author Spencer Gibb
+ */
+public class CachedDiscoveryClient implements ScopedDiscoveryClient {
```

Comment:
javadoc the impact of cached things.. for example, I'm guessing this means the instance list will be available when the backend is down, but might be stale. Is there anything in the backend to invalidate cache entries? If so, note behaviors folks can rely on. otherwise, don't